src/main/java/com/angelbroking/smartapi/SmartConnect.java

```diff
@@ -704,11 +703,28 @@ public String candleData(JSONObject params) {
 		}
 	}
 
+	/**
+	 * Get Market Data.
+	 *
+	 * @param params is market data params.
+	 * @return returns the details of market data.
+	 */
+	public JSONObject marketData(JSONObject params) {
+		try{
+			String url = routes.get("api.market.data");
+			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
+			return response.getJSONObject("data");
+		}catch (Exception | SmartAPIException e) {

```

Comment:
what is need of calling SmartAPIException while you are already calling Exception?

Reply:
The SmartAPIException class catches custom messages and custom error codes.
 also, it extends to the Throwable class. So need to catch it separately